internal/provider/resource_hvn_route.go

```diff
@@ -1,14 +1,293 @@
+			HvnConnection: targetLink,
+		},
+	}
+	log.Printf("[INFO] Creating HVN route for HVN (%s) with destination CIDR %s", hvnLink.ID, destination)
+	hvnRouteResp, err := client.Network.CreateHVNRoute(hvnRouteParams, nil)
+	if err != nil {
+		return diag.Errorf("unable to create HVN route for HVN (%s) with destination CIDR %s: %v", hvnLink.ID, destination, err)
+	}
+
+	hvnRoute := hvnRouteResp.Payload.Route
+
+	// Set the globally unique id of this HVN route in the state now since it has
+	// been created, and from this point forward should be deletable.
+	link := newLink(hvnRoute.Hvn.Location, HVNRouteResourceType, hvnRoute.ID)
+	url, err := linkURL(link)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(url)
+
+	// Wait for HVN route to be created.
+	if err := clients.WaitForOperation(ctx, client, "create HVN route", loc, hvnRouteResp.Payload.Operation.ID); err != nil {
+		return diag.Errorf("unable to create HVN route for HVN (%s) with destination CIDR %s:%v", hvnLink.ID, destination, err)
+	}
+
+	log.Printf("[INFO] Created HVN route for HVN (%s) with destination CIDR %s", hvnRoute.ID, hvnRoute.Hvn.ID)
+
+	hvnRoute, err = clients.WaitForHVNRouteToBeActive(ctx, client, destination, hvnLink.ID, loc, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := setHVNRouteResourceData(d, hvnRoute, loc); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceHvnRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+
+	hvnLink, err := parseLinkURL(hvn, HvnResourceType)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	destination := d.Get("destination_cidr").(string)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	route, err := clients.ListHVNRoutes(ctx, client, hvnLink.ID, destination, "", "", loc)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			log.Printf("[WARN] HVN route for HVN (%s) with destination CIDR %s not found, removing from state", hvnLink.ID, destination)
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("unable to retrieve HVN route for HVN (%s) with destination CIDR %s: %v", hvnLink.ID, destination, err)
+	}
+
+	if len(route) != 1 {
+		return diag.Errorf("Unexpected number of HVN routes returned when waiting for route with destination CIDR of %s for HVN (%s) to be Active: %d", destination, hvnLink.ID, len(route))
+	}
+
+	// The HVN route failed to provision properly so we want to let the user know and
+	// remove it from state.
+	if route[0].State == networkmodels.HashicorpCloudNetwork20200907HVNRouteStateFAILED {
+		log.Printf("[WARN] HVN route for HVN (%s) with destination CIDR %s failed to provision, removing from state", hvnLink.ID, destination)
```

Comment:
@bcmdarroch same question here, why don't we just fail the tf apply?

Reply:
Same answer, this updates the TF state during a `terraform refresh`.